interview-questions.md

```diff
@@ -8,41 +8,45 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+Your answer: A parameter is a variable that is only in the scope of a function and is not accessible within a global scope. An argument is when there is a need to change a variable from one state to another.
 
-  Researched answer:
+Researched answer: Parameters are variables listed as a part of function declaration. Arguments are values passed to the function when it is invoked. The difference between the two is the argument is the actual value and the parameters are just the names listed in the function’s definition.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+Your answer: The .map method takes in indexes, values, and arrays. Each of these parameters are used in order to provide logic and function to arrays. Only the indexes are required and the rest are optional.
 
-  Researched answer:
+Researched answer: The .map function uses the predefined parameters: current element/value, the index of the current element, and the whole array. The current element/value is required and the index and array arguments are optional.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+Your answer: The map is used to compare and change the same amount of indexes in an array. A filter is used to compare, change, and add/delete to an array. The biggest difference is that a filter will use conditionals to return a different amount of indexes in an array. A map will return the same amount of indexes.
 
-  Researched answer:
+
+Researched answer: Both map and filter are higher order functions. The filter method creates a new array with array elements that pass a test implemented from the provided function. The map method also creates a new array with the results of calling every array element. The map method allows items to be manipulated when returned.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+Your answer: A function requires a return. A function can input parameter values to manipulate data using conditionals and iterating through the input. A method is similar to a function that can manipulate data and return an input.
 
-  Researched answer:
+Researched answer: A function and method are a set of instructions, but a method is associated with an object. Functions are like recipes and execute a set of instructions on data or variables and return the result. A method is similar in that it can perform a task, but the difference is a method is associated with an object and a function is not. Some examples of built-in methods are toLowerCase() and toUpperCase().
```

Comment:
This answer is starting to look very similar to some documentation.  Make sure that you are forming this in your own words and not rewording documentation.